include/biscuit/cpuinfo.hpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c), 2022, KNS Group LLC (YADRO)
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+#pragma once
+
+#include <biscuit/assembler.hpp>
+#include <biscuit/registers.hpp>
+#include <cstddef>
+#include <cstdint>
+
+#if defined(__linux__) && defined(__riscv)
+#include <sys/auxv.h>
+#include <sys/prctl.h>
+#include <asm/hwcap.h>
+#endif
+
+namespace biscuit {
+
+#ifndef COMPAT_HWCAP_ISA_I
+#define COMPAT_HWCAP_ISA_I  (1U << ('I' - 'A'))
```

Comment:
out of curiosity, where would COMPAT_HWCAP_ISA_I (and the others be defined from that needs the ifdef before the defines)?

I was thinking these should be undefined after the enum class, so that this header isn't potentially introducing global defines, but if there's a header that can have existing defines it's probably best to not do that

Reply:
Sorry, I should have clarified it in the PR description.

These variables are defined in [hwcap.h](https://github.com/torvalds/linux/blob/master/arch/riscv/include/uapi/asm/hwcap.h) header file from the RISC-V linux kernel.
There is no ```COMPAT_HWCAP_ISA_V``` there since RVV support has not been officially added to the kernel yet. So we use a [patched](https://patchwork.kernel.org/project/linux-riscv/cover/20220921194629.1480202-1-stillson@rivosinc.com/) kernel.